apps/meteor/app/lib/server/methods/getMessages.ts

```diff
@@ -16,9 +16,22 @@ Meteor.methods({
 
 		const msgs = Messages.findVisibleByIds(messages).fetch() as IMessage[];
 		const rids = [...new Set(msgs.map((m) => m.rid))];
+		const prids = [
+			...new Set(
+				rids.reduce<string[]>((prids, rid) => {
+					return Rooms.findOneById(rid).then((room: IRoom) => {
```

Comment:
`findOneById` can be used as a synchronous method here

Reply:
updated!